programs/voter-stake-registry/src/instructions/create_deposit_entry.rs

```diff
@@ -61,7 +61,6 @@ pub fn create_deposit_entry(
     deposit_entry_index: u8,
     kind: LockupKind,
     period: LockupPeriod,
-    delegate: Pubkey,
```

Comment:
Wdyt of keeping the delegate here? This way you'll not have to update it on every stake related operation, but only check.